backend/src/v2/driver/driver.go

```diff
@@ -1861,45 +1869,36 @@ func resolveUpstreamParameters(cfg resolveUpstreamOutputsConfig) (*structpb.Valu
 
 			glog.V(4).Infof("Deserialized dagOutputParametersMap: %v", dagOutputParametersMap)
 
+			// For this section, if the currentTask we are looking for is within
+			// a ParallelFor DAG, this means the actual task that produced the
+			// output we need has multiple iterations so we have to gather all
+			// them and fan them in by collecting them into a list i.e.
+			// kfp.dsl.Collected support.
+			starterTaskQueue, err := isCollection(cfg, currentTask)
```

Comment:
How about something like this? It's not actually a "queue" until `CollectInputs` is called.
```suggestion
			parallelForTasks, err := isCollection(cfg, currentTask)
```